donkey.lua

```diff
@@ -27,6 +28,29 @@ end
 local loadSize   = {3, opt.imageSize, opt.imageSize}
 local sampleSize = {3, opt.cropSize, opt.cropSize}
 
+if opt.imgExtInsensitive then
+   -- Overwrite image.load()
+   -- Load an image (with image.load()) regardless its extension
+   function image.load(p, n, t)
+      local img, ok
+      local f = io.open(p, 'r')
+      if f then
+         local h = f:read(4)
+         f:close()
+
+         if h:startswith(string.char(0xff, 0xd8, 0xff)) then
```

Comment:
@apaszke that's a detail but if we want to avoid requiring Penlight for a single function (`startswith`) - since torch/image does not use it right now - we can as well use [`string.find`](https://github.com/stevedonovan/Penlight/blob/16d1493/lua/pl/stringx.lua#L85-L87) directly.